### ISSUE TITLE
False negative in inheritance check with `type[]`

### ISSUE THREAD

**Issue by JelleZijlstra** [microsoft/pyright#3309]
Pyright (in latest pylance) reports no errors for:

```python
from typing import Any, Generic, TypeVar

T = TypeVar("T")


class Foo(Generic[T]):
    y: type[T]


class SubFoo(Foo[Any]):
    y = 42
```

But 42 is not a `type[Any]`.

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release.

**Comment by erictraut** [comment]
This is fixed in pyright 1.1.236, which I just published. It will also be included in the next release of pylance.
